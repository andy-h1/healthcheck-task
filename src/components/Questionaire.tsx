import { QuestionType } from "../types/questionaire";
import { ChevronRightIcon } from "@heroicons/react/16/solid";

type QuestionaireProps = {
  question: QuestionType;
  selectedAnswerId: string | null;
  isAnswerSelected: boolean;
  onAnswerSelect: (answerId: string) => void;
  onNext: () => void;
};

const Questionaire = ({
  question,
  selectedAnswerId,
  isAnswerSelected,
  onAnswerSelect,
  onNext
}: QuestionaireProps) => {
  return (
    <div className="flex h-2/3 w-full flex-col items-center justify-between px-3 ">
      <div>
        <h1 className="text-2xl font-extrabold text-gray-600">
          {question.question_text}
        </h1>
        <span className="my-10 flex justify-center">
          {question.answers.map((answer) => (
            <button
              key={answer.id}
              onClick={() => onAnswerSelect(answer.id)}
              className={`mx-3 inline-flex w-1/2 items-center justify-center rounded-full px-8 py-4 text-sm font-bold shadow-sm ring-1 ring-inset ring-gray-300
                        ${
                          answer.id === selectedAnswerId
                            ? "bg-teal-400 text-white"
                            : "bg-white text-teal-400"
                        }`}
            >
              {answer.label}
            </button>
          ))}
        </span>
      </div>
      <button
        onClick={onNext}
        disabled={!isAnswerSelected}
        className={`my-4 inline-flex w-full items-center justify-center  ${
          isAnswerSelected ? "bg-teal-500 hover:bg-teal-700" : "bg-gray-300"
        } rounded px-4 py-2 font-bold text-white`}
      >
        Next
        <ChevronRightIcon className="h-5 w-5" aria-hidden="true" />
      </button>
    </div>
  );
};

export default Questionaire;
