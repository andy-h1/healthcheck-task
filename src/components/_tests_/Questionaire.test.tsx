import { render, screen } from "@testing-library/react";
import { describe, test, expect, vi } from "vitest";
import Questionaire from "../Questionaire";

const props = {
  question: {
    id: "heartburn_previous_treatment",
    question_text: "Have you received any treatment for heartburn previously?",
    answers: [
      { id: "heartburn_previous_treatment_yes", label: "Yes", score: 0 },
      { id: "heartburn_previous_treatment_no", label: "No", score: 0 }
    ],
    next: [{ next_question: "heartburn_weekly_burns" }]
  },
  selectedAnswerId: "",
  isAnswerSelected: false,
  onAnswerSelect: vi.fn(),
  onNext: vi.fn()
};

describe("Questionaire component", () => {
  test("it should render", () => {
    const { container } = render(<Questionaire {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test("Next button should be disabled on render", () => {
    render(<Questionaire {...props} />);

    const nextButton = screen.getByRole("button", { name: /next/i });
    expect(nextButton).toBeDisabled();
  });
});
