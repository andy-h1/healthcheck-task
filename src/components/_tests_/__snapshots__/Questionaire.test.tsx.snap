// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Questionaire component > it should render 1`] = `
<div
  class="flex h-2/3 w-full flex-col items-center justify-between px-3 "
>
  <div>
    <h1
      class="text-2xl font-extrabold text-gray-600"
    >
      Have you received any treatment for heartburn previously?
    </h1>
    <span
      class="my-10 flex justify-center"
    >
      <button
        class="mx-3 inline-flex w-1/2 items-center justify-center rounded-full px-8 py-4 text-sm font-bold shadow-sm ring-1 ring-inset ring-gray-300
                        bg-white text-teal-400"
      >
        Yes
      </button>
      <button
        class="mx-3 inline-flex w-1/2 items-center justify-center rounded-full px-8 py-4 text-sm font-bold shadow-sm ring-1 ring-inset ring-gray-300
                        bg-white text-teal-400"
      >
        No
      </button>
    </span>
  </div>
  <button
    class="my-4 inline-flex w-full items-center justify-center  bg-gray-300 rounded px-4 py-2 font-bold text-white"
    disabled=""
  >
    Next
    <svg
      aria-hidden="true"
      class="h-5 w-5"
      data-slot="icon"
      fill="currentColor"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M6.22 4.22a.75.75 0 0 1 1.06 0l3.25 3.25a.75.75 0 0 1 0 1.06l-3.25 3.25a.75.75 0 0 1-1.06-1.06L8.94 8 6.22 5.28a.75.75 0 0 1 0-1.06Z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
